### PR TITLE
[Contract Sync Job] Support Contribution upsert

### PR DESCRIPTION
**Pull request by amrro**
## Linear Ticket
[PRO-535 - Verify contract sync job is working](https://linear.app/govrn/issue/PRO-535/verify-contract-sync-job-is-working)

## Description
To avoid duplicates, contribution will be upserted based on: `chain_id` & `on_chain_id`.

### CODE REVIEW

File: apps/contract-sync-job/src/db.ts [hunk 1]
```diff
@@ -63,11 +79,71 @@ export async function getContribution(
...
+        connect: { name: 'minted' },
+      },
+      on_chain_id: contribution.on_chain_id,
+      chain: { connect: { chain_id: `${contribution.chain_id}` } },
+    },
+    update: {
+      name: { set: contribution.name },
+      on_chain_id: { set: contribution.on_chain_id },
+      // proof: { set: contribution.proof },
+      // details: { set: contribution.details },
```
Comment (alexkeating) [inline apps/contract-sync-job/src/db.ts:112]: Why comment these out?

File: apps/contract-sync-job/src/main.ts [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
 const JOB_NAME = 'contract-sync-job';
 
 const CONTRACT_ADDRESS = process.env.CONTRACT_ADDRESS;
 const CHAIN_URL = process.env.CHAIN_URL;
+const CHAIN_ID = 100;
 
 const networkConfig: NetworkConfig = {
   address: CONTRACT_ADDRESS,
-  chainId: 100,
+  chainId: CHAIN_ID,
```
Comment (alexkeating) [inline apps/contract-sync-job/src/main.ts:26]: Nice

File: apps/contract-sync-job/src/db.ts (hunk 1 above)
Comment (amrro) [reply apps/contract-sync-job/src/db.ts]: I fixed it: 56aca17cb39c0b83874bd426ab8618b83dbb834b

Prisma accepts only `string | null` values, but it crashes when the value is `undefined`.

**Comment by alexkeating** [comment]
> @alexkeating Is it feasible include `tx_hash` to `GovrnGraphClient.listContributions()`'s return values?

Yea, that is feasible and fine

File: libs/govrn-subgraph-client/src/lib/subgraph/queries/queries.graphql [hunk 3]
```diff
@@ -51,26 +55,3 @@ query listContributions(
     ...contributionFields
   }
 }
-
-fragment guildFields on Guild {
```
Comment (Flip-Liquid) [inline libs/govrn-subgraph-client/src/lib/subgraph/queries/queries.graphql:55]: is this related to the PR?

File: apps/protocol-api/src/main.ts [hunk 4]
```diff
@@ -153,6 +155,13 @@ const permissions = shield(
       name: or(isAuthenticated, hasToken),
       users: or(isAuthenticated, hasToken),
     },
+    Chain: {
+      id: hasToken,
```
Comment (Flip-Liquid) [inline apps/protocol-api/src/main.ts:159]: why no isAuthenticated?

File: libs/govrn-subgraph-client/src/lib/subgraph/queries/queries.graphql (hunk 3 above)
Comment (alexkeating) [reply libs/govrn-subgraph-client/src/lib/subgraph/queries/queries.graphql:55]: Yea, we need to add fields to the subgraph and this prevented us from adding those fields

File: apps/protocol-api/src/main.ts (hunk 4 above)
Comment (alexkeating) [reply apps/protocol-api/src/main.ts:159]: Because it is not used by the frontend, best to only add permissions when needed.
